src/java.base/share/classes/java/lang/foreign/SymbolLookup.java

```diff
@@ -144,9 +144,39 @@ public interface SymbolLookup {
      * @param name the symbol name
      * @return a zero-length memory segment whose address indicates the address of
      *         the symbol, if found
+     * @see #findOrThrow(String)
      */
     Optional<MemorySegment> find(String name);
 
+    /**
+     * {@return the address of the symbol with the provided {@code name} or throws an
```

Comment:
I suggest using the same javadoc structure as the main `find` method. Note that there we have a summary, then a more detailed `@return` which talks about zero length memory segments. This should do the same.